libyul/optimiser/LoadResolver.cpp

```diff
@@ -76,3 +102,44 @@ void LoadResolver::tryResolve(
 			if (inScope(*value))
 				_e = Identifier{locationOf(_e), *value};
 }
+
+void LoadResolver::tryEvaluateKeccak(
+	Expression& _e,
+	std::vector<Expression> const& _arguments
+)
+{
+	auto valueOfIdentifier = [&](YulString const& _name) -> optional<u256>
+	{
+		if (m_value.count(_name))
+			if (Literal const* literal = get_if<Literal>(m_value.at(_name).value))
+				return valueOfLiteral(*literal);
+		return nullopt;
+	};
+
+	yulAssert(_arguments.size() == 2, "");
+	Identifier const* memoryKey =  std::get_if<Identifier>(&_arguments.at(0));
+	Identifier const* offset = std::get_if<Identifier>(&_arguments.at(1));
+
+	if (!memoryKey || !offset) return;
```

Comment:
Not sure if our coding guidelines expect a return on the next line

Reply:
I think I never saw it on the same line in any case...

Reply:
Fixed.